compiler/passes/parallel.cpp

```diff
@@ -659,43 +679,52 @@ static void create_block_fn_wrapper(FnSymbol* fn, CallExpr* fcall, BundleArgsFnD
   // Create a call to the original function
   CallExpr *call_orig = new CallExpr(fn);
   bool first = true;
+  int i = 0;
   for_fields(field, ctype)
   {
-    // insert args
-    VarSymbol* tmp = newTemp(field->name, field->qualType());
-    wrap_fn->insertAtTail(new DefExpr(tmp));
-    wrap_fn->insertAtTail(
-        new CallExpr(PRIM_MOVE, tmp,
-        new CallExpr(PRIM_GET_MEMBER_VALUE, wrap_c, field)));
-
-    // Special case: 
-    // If this is an on block, remember the first field,
-    // but don't add to the list of actuals passed to the original on_fn.
-    // It contains the locale on which the new task is launched.
-    if (first && fn->hasFlag(FLAG_ON))
-      /* no-op */;
-    else
-      call_orig->insertAtTail(tmp);
+    // Skip runtime header
+    if (i > 0) {
```

Comment:
that doesn't work well because fields.first is an Expr* while field is a Symbol*. Besides that, the general pattern of if ( something ) continue; here is challenging because the i++ still needs to happen (for the next loop anyway). I think the conditional is actually the simplest way to write the correct solution.